module/zfs/metaslab.c

```diff
@@ -268,14 +366,31 @@ metaslab_class_validate(metaslab_class_t *mc)
 	return (0);
 }
 
-void
+static void
 metaslab_class_space_update(metaslab_class_t *mc, int64_t alloc_delta,
     int64_t defer_delta, int64_t space_delta, int64_t dspace_delta)
 {
 	atomic_add_64(&mc->mc_alloc, alloc_delta);
 	atomic_add_64(&mc->mc_deferred, defer_delta);
 	atomic_add_64(&mc->mc_space, space_delta);
 	atomic_add_64(&mc->mc_dspace, dspace_delta);
+
+	spa_t *spa = mc->mc_spa;
+	uint64_t allocated = mc->mc_alloc;
+
+	if (mc == spa_normal_class(spa)) {
+		MS_CLASS_STAT(normal_allocated) = allocated;
+		if (allocated > MS_CLASS_STAT(normal_highest_allocated))
+			MS_CLASS_STAT(normal_highest_allocated) = allocated;
```

Comment:
So, the `_allocated` kstat is the amount of space currently allocated in this class, and `_highest_allocated` is the fullest this class has been since the pool was imported?  Those seem like weird things to have kstats for, though I can't really put my finger on why.  Did we have kstats for how full the pool (or specific devices) were before?  I guess kstats more commonly record an amount of events (e.g. # bytes allocated, # allocations).

Is there a way to find out how full the class is from the `zpool` CLI?  Is that using this kstat or something else?  Seems like it should use a ZFS-specific mechanism (like we do for vdev/pool fullness today).

Reply:
these kstats were incomplete and have been removed from this pull-request.